### PR TITLE
#11 Nesting Workflows

### LINKED ISSUE #11: Nesting Workflows
This issue addresses the content of the Nesting Workflows page. This page aims to show the user how to create reusable workflows and nest them inside a Workflow node. It includes how to: 
- Define workflow inputs and output with nodes
- Create a Workflow node
- Execute workflow node.
- Add link to Workflow Process Nodes

Screenshots should be added where needed

### PR DESCRIPTION
**Pull request by juliavisserinbibo**
## Issue
Closes #11

## Description
This PR addresses the content of the Nesting Workflows page. This page aims to show the user how to create reusable workflows and nest them inside a Workflow node. It includes how to:
- Define workflow inputs and output with nodes
- How to use Workflow node

This page includes a link to the Workflow Node page and the Workflow Process Node page, so the user can look for more specific information. This pages are still not created.

### CODE REVIEW

File: docs/getting_started/basics/nesting_workflows.md [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+The `Workflow` node allows to embed the execution of a workflow inside another workflow. This node has a *file* plug, where the workflow to nest is to be indicated. When the file plug content is set to reference a valid workflow file (with a *.sft* extension), the node will automatically update by creating the input and output plugs corresponding to the sourced workflow inputs and outputs. Then, when the `Workflow` node gets executed, it will execute the referenced workflow. 
 
-![Nested Workflows](../images/nested_workflows.gif)
+![Nested Workflows](../../images/workflow_nesting.gif)
+
+To learn more about the `Workflow` node functionalities, please refer to the [`Workflow`](../../nodes/workflow.md) documentation.
+
+### Workflows on External Processes
+
+When using the `Workflow` node, the execution of the workflow happens on the same python process Shift is running in. However, its possible to define and execute a workflow in an external process. To do so, please refer to the [`Workflow Process`](../../nodes/workflow_process.md) documentation.
```
Comment (albertochimentiinbibo) [inline docs/getting_started/basics/nesting_workflows.md:22]:
```suggestion
When using the `Workflow` node, the execution of the workflow happens on the same python process Shift is running in. However, its possible to define and execute a workflow in an external process or with a different Python interpreter. To do so, please refer to the [`Workflow Process`](../../nodes/workflow_process.md) documentation.
```

**Review by albertochimentiinbibo — CHANGES_REQUESTED** [review]

File: docs/getting_started/basics/nesting_workflows.md [hunk 2]
```diff
@@ -1,16 +1,22 @@
 # Nesting Workflows
 
-In Shift any workflow can be added and executed as an operator.
+In Shift, any workflow can be embedded into a node and executed within another workflow.
 
 ## Setting Workflow I/Os
+To create a reusable workflow, the input and output of the workflow must be specified. This is done through `Input` and `Output` nodes. The plugs created in these nodes define which inputs are expected by the workflow and which are the outputs that will be returned.
```
Comment (leegeertseninbibo) [inline docs/getting_started/basics/nesting_workflows.md:6]:
```suggestion
To create a reusable workflow, the inputs and outputs of the workflow must be specified. This is done through `Input` and `Output` nodes. The `through` plugs created in these nodes define which inputs are expected by the workflow and which are the outputs that will be returned.
```

**Review by leegeertseninbibo — CHANGES_REQUESTED** [review]

File: docs/getting_started/basics/nesting_workflows.md (hunk 1 above)
Comment (hectorjimenezinbibo) [inline docs/getting_started/basics/nesting_workflows.md:22]: Remember to write Python with the P in uppercase 👍 
```suggestion
When using the `Workflow` node, the execution of the workflow happens on the same Python process Shift is running in. However, its possible to define and execute a workflow in an external process. To do so, please refer to the [`Workflow Process`](../../nodes/workflow_process.md) documentation.
```

**Review by hectorjimenezinbibo — CHANGES_REQUESTED** [review]

File: docs/getting_started/basics/nesting_workflows.md (hunk 1 above)
Comment (juliavisserinbibo) [reply docs/getting_started/basics/nesting_workflows.md:22]: It's is not an "or" right? When launching a workflow with another python interpreter you are executing the workflow in an external process

File: docs/getting_started/basics/nesting_workflows.md (hunk 2 above)
Comment (juliavisserinbibo) [reply docs/getting_started/basics/nesting_workflows.md:6]: Should we link this concept somewhere? I feel it's a bit out of the blue to mention its a through plug

**Review by JorgeHernandezibanezinbibo — APPROVED** [review]

**Review by hectorjimenezinbibo — APPROVED** [review]

File: docs/getting_started/basics/nesting_workflows.md (hunk 2 above)
Comment (leegeertseninbibo) [reply docs/getting_started/basics/nesting_workflows.md:7]: I think we can mention it in this documentation as we mostly use through plugs for input and output operators

Comment (juliavisserinbibo) [reply docs/getting_started/basics/nesting_workflows.md]: I changed the sentence

File: docs/getting_started/basics/nesting_workflows.md (hunk 1 above)
Comment (albertochimentiinbibo) [reply docs/getting_started/basics/nesting_workflows.md]: It's an external process handled by a Python interpreter which can be the same one as the current session of Shift or a different one. I think the important feature to talk about here is the fact that you can set a different interpreter, not that it runs in a subprocess. Also as the Python execution is single threaded, there is no gain in executing in a subprocess but the fact that you can change intepreter... as you still have to wait for the process to complete execution before being able to continue. I think we can probably rephrase the statement?

It is not an xor xD

Comment (juliavisserinbibo) [reply docs/getting_started/basics/nesting_workflows.md]: What do you think about:
"However, it's possible to define and execute a workflow in an external process running in a different Python interpreter."

Comment (juliavisserinbibo) [reply docs/getting_started/basics/nesting_workflows.md]: Done!

File: docs/getting_started/basics/nesting_workflows.md [hunk 3]
```diff
@@ -1,16 +1,23 @@
 # Nesting Workflows
 
-In Shift any workflow can be added and executed as an operator.
+In Shift, any workflow can be embedded into a node and executed within another workflow.
 
 ## Setting Workflow I/Os
 
-First of all we need to tell Shift which are the *Inputs* and *Outputs* of the workflow in question. To do that we make use of special operators called `Input` and `Output`. Then we need to create plugs for these operators corresponding to what we expect to be requested by the workflow and what we want to be returned.
+To create a reusable workflow, the input and output of the workflow must be specified. This is done through `Input` and `Output` nodes. The plugs created in these nodes, which must be `through` plugs, define which inputs are expected by the workflow and which are the outputs that will be returned.
```
Comment (albertochimentiinbibo) [inline docs/getting_started/basics/nesting_workflows.md:7]:
```suggestion
To create a reusable workflow, the input and output of the workflow must be specified. This is done through `Input` and `Output` nodes. The plugs created in these nodes must be `through` plugs and define which inputs are expected by the workflow and which are the outputs that will be returned.
```

**Review by albertochimentiinbibo — CHANGES_REQUESTED** [review]

**Review by albertochimentiinbibo — APPROVED** [review]

**Review by leegeertseninbibo — APPROVED** [review]

**Pull request merged**